numpy/lib/function_base.py

```diff
@@ -989,7 +989,17 @@ def average(a, axis=None, weights=None, returned=False):
     TypeError: Axis must be specified when shapes of a and weights differ.
 
     """
-    a = np.asanyarray(a)
+    # replace the next few lines with "a = np.asanyarray(a)" in the future
+    if (type(a) not in (np.ndarray, np.matrix) and
+            issubclass(type(a), np.ndarray)):
+        warnings.warn("np.average currently does not presevre subclasses, but "
+                      "will do so in the future to match the behaior of most "
+                      "other numpy functions such as np.mean. In particular, "
+                      "this means calls which returned a scalar may return a "
+                      "0-d subclass object instead.", FutureWarning)
```

Comment:
Since I hope to change this everywhere else as well by 1.12 (and probably force it in the numpy code base for the most part), could you add a stacklevel=2?